### PR TITLE
Method uDSR for competition

### PR DESCRIPTION
**Pull request by brendenpetersen**
Competition Checklist:
- [x] title of this PR is meaningful, i.e. "method X for comp"
- [x] A folder has been added to `submission/` with a meaningful name corresponding to your method name.
- The added folder includes these elements:
    - [x] `metadata.yml` (**required**): A file describing your submission, following the descriptions in `example/metadata.yml`.  
    - [x] `regressor.py` (**required**): a Python file that defines your method, named appropriately. See [submission/feat-example/regressor.py](https://github.com/cavalab/srbench/blob/Competition2022/submission/feat-example/regressor.py) for complete documentation.  It contains:
        - [x]  `est`: a sklearn-compatible `Regressor` object. 
        - [x]  `model(est, X=None)`: a function that returns a [**sympy-compatible**](https://www.sympy.org) string specifying the final model. It can optionally take the training data as an input argument. See [guidance below](###-returning-a-sympy-compatible-model-string). 
        - [x]  `eval_kwargs` (optional): a dictionary that can specify method-specific arguments to `evaluate_model.py`.
        
    - [x] `LICENSE` *(optional)* A license file
    - [x] `environment.yml` *(optional)*: a [conda environment file](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file) that specifies dependencies for your submission. 
    - [x] `install.sh` *(optional)*: a bash script that installs your method. 
    - [x] additional files *(optional)*: you may include a folder containing the code for your method in the submission. 

I have verified that:

- [x] install scripts do not require sudo permissions. 
- [x] if pulled remotely, the source code is a fixed version (i.e., rerunning `install.sh` shouldn't pulll a different version of the code when run multiple times.)

### CODE REVIEW

**Comment by brendenpetersen** [comment]
We are submitting now to ensure there are no installation/test issues. But we would still like to make a few more changes in the Python for things like timeout handling which should not affect installation. Thanks for all of your help!

**Comment by brendenpetersen** [comment]
@lacava I'm not sure the installation issue. I don't really use conda. Using the `environment.yml` file and subsequence `install.sh` works on our machines. Our method requires `python3.6.x` or `python3.7.x` and requires `gcc3.1.x`, if that is helpful.

File: submission/uDSR/environment.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: udsr
```
Comment (lacava) [inline submission/uDSR/environment.yml:1]: I suggest removing as much of the package versioning as you can to help conda find a matching set of dependencies.

Comment (brendenpetersen) [reply submission/uDSR/environment.yml:1]: Will try that, thanks. Working on paring down a minimal dependency version. Some of the conflicts seem odd though, e.g. `package argon2-cffi-bindings-21.2.0-py39hb9d737c_2 requires python >=3.9,<3.10.0a0`, we don't have `argon2` on our dependency list -- is it on the SRBench dependency list? Is the base environment trying to combine envs from all the other submissions?

Comment (foolnotion) [reply submission/uDSR/environment.yml:1]: I think you might need to use
```yml
channels:
    - conda-forge
```
Removing versions should help a lot.

Comment (lacava) [reply submission/uDSR/environment.yml:1]: the environments of each submission are separate, but your code has to play well with the srbench dependencies [here](https://github.com/cavalab/srbench/blob/Competition2022/environment.yml)

not sure of `argon2-cffi`

Comment (brendenpetersen) [reply submission/uDSR/environment.yml:1]: @lacava Those linked dependencies work fine.

@foolnotion Thanks, I will add `conda-forge` and remove lots of versions. I don't know how to test this though other than just updating the PR and crossing my fingers...

Comment (brendenpetersen) [reply submission/uDSR/environment.yml:1]: @lacava It just failed with a dirt-simple environment file consisting of:

name: udsr
channels:
  - defaults
  - conda-forge
dependencies:
  - python=3.6
```

Seems like there is some other stuff in the base `srcomp`?

Comment (foolnotion) [reply submission/uDSR/environment.yml:1]: Could you try removing the `defaults` channel or putting it after `conda-forge`? Packages from different channels don't mix well.

Comment (brendenpetersen) [reply submission/uDSR/environment.yml:1]: @foolnotion Done. Same error. Is this issue possibly unrelated to my submission?

Comment (brendenpetersen) [reply submission/uDSR/environment.yml:1]: @lacava @foolnotion I mucked around with the versions of things until it worked. I guess the base `srcomp` really does not like `python3.6` for some reason. But it's working for me now with `python3.7`. Thanks again for your help and prompt replies :)

**Comment by brendenpetersen** [comment]
@lacava Tests are passing now but I'll make a few changes before merging if that's okay, thanks!

**Comment by lacava** [comment]
thanks for the submission!

**Comment by brendenpetersen** [comment]
Thanks @lacava this is good to go. (Just FYI since I don't see it in `official_competitors` yet.)

**Pull request merged**